r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -278,7 +279,7 @@ public void start()
       long timeOut = Math.min(_idleTimeout, _waitTimeout);
       if (timeOut > 0)
       {
-        long freq = Math.min(timeOut / 10, 1000);
+        long freq = Math.min(timeOut / 10, 50);
```

Comment:
What I meant here was to set an upper and lower bound like in   Math.max(50,Math.min(timeOut / 10, 1000));